godot-core/src/builtin/color.rs

```diff
@@ -4,30 +4,511 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
  */
 
+use crate::builtin::inner::InnerColor;
+use crate::builtin::GodotString;
 use godot_ffi as sys;
+use std::ops;
 use sys::{ffi_methods, GodotFfi};
 
+/// Color built-in type, in floating-point RGBA format.
+///
+/// Channel values are _typically_ in the range of 0 to 1, but this is not a requirement, and
+/// values outside this range are explicitly allowed for e.g. High Dynamic Range (HDR).
 #[repr(C)]
-#[derive(Copy, Clone, Debug, PartialEq)]
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd)]
 pub struct Color {
+    /// The color's red component.
     pub r: f32,
+    /// The color's green component.
     pub g: f32,
+    /// The color's blue component.
     pub b: f32,
+    /// The color's alpha component. A value of 0 means that the color is fully transparent. A
+    /// value of 1 means that the color is fully opaque.
     pub a: f32,
 }
 
 impl Color {
-    pub fn new(r: f32, g: f32, b: f32, a: f32) -> Self {
+    // TODO implement all the other color constants using code generation
+
+    /// Transparent black.
+    pub const TRANSPARENT_BLACK: Color = Self::new(0.0, 0.0, 0.0, 0.0);
+
+    /// Transparent white. Equivalent of `Color.TRANSPARENT` in GDScript.
+    pub const TRANSPARENT_WHITE: Color = Self::new(1.0, 1.0, 1.0, 0.0);
+
+    /// Opaque black.
+    pub const BLACK: Color = Self::new(0.0, 0.0, 0.0, 1.0);
+
+    /// Opaque white.
+    pub const WHITE: Color = Self::new(1.0, 1.0, 1.0, 1.0);
+
+    /// Constructs a new `Color` with the given components.
+    pub const fn new(r: f32, g: f32, b: f32, a: f32) -> Self {
         Self { r, g, b, a }
     }
+
+    /// Constructs a new `Color` with the given components. More explicit alias of [`Color::new`].
+    pub const fn from_rgba(r: f32, g: f32, b: f32, a: f32) -> Self {
+        Self::new(r, g, b, a)
+    }
+
+    /// Constructs a new `Color` with the given color components, and the alpha channel set to 1.
+    pub const fn from_rgb(r: f32, g: f32, b: f32) -> Self {
+        Self::from_rgba(r, g, b, 1.0)
+    }
+
+    /// Constructs a new `Color` with the given components as bytes. 0 is mapped to 0.0, 255 is
+    /// mapped to 1.0. Equivalent of the global constructor function `Color8` in GDScript.
+    pub fn from_rgba8(r: u8, g: u8, b: u8, a: u8) -> Self {
+        Self::from_rgba(from_u8(r), from_u8(g), from_u8(b), from_u8(a))
+    }
```

Comment:
> Equivalent of the global constructor function `Color8` in GDScript.

For dicts and arrays, I started introducing some sort of "standard notation" to quickly see how Rust symbols map to GDScript ones, here are some examples (last doc line in italics):

https://github.com/godot-rust/gdextension/blob/da04ee1efc7dac444473213177320d9b20986962/godot-core/src/builtin/dictionary.rs#L71-L80

https://github.com/godot-rust/gdextension/blob/da04ee1efc7dac444473213177320d9b20986962/godot-core/src/builtin/dictionary.rs#L113-L121

Do you think this would be applicable here, too? You could still mention "global constructor" for clarity.

Reply:
Maybe we can also not use italics, and instead put the "Godot equivalent:" part in bold:
```rs
/// **Godot equivalent:** `dict.get(key, null)`
```

Might render better, would need to check.

Reply:
I like that. Done.